controllers/nstemplatetier/nstemplatetier_controller_test.go

```diff
@@ -517,79 +856,201 @@ func TestReconcile(t *testing.T) {
+				assert.Equal(t, reconcile.Result{}, res) // no explicit requeue
+			})
+			t.Run("Spaces", func(t *testing.T) {
+				// given
+				previousBasicTier := tiertest.BasicTier(t, tiertest.PreviousBasicTemplates)
+				basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates, tiertest.WithCurrentUpdateInProgress())
+				initObjs := []runtime.Object{basicTier}
+				initObjs = append(initObjs, spacetest.NewSpaces(10, "user-%d", spacetest.WithTierNameAndHashLabelFor(previousBasicTier))...)
+
+				r, req, cl := prepareReconcile(t, basicTier.Name, initObjs...)
+				cl.MockGet = func(ctx context.Context, key types.NamespacedName, obj client.Object) error {
+					if _, ok := obj.(*toolchainv1alpha1.TemplateUpdateRequest); ok {
+						return fmt.Errorf("mock error") // must not be a `NotFoundError` in this test
+					}
+					return cl.Client.Get(ctx, key, obj)
+				}
+				// when
+				res, err := r.Reconcile(context.TODO(), req)
+				// then
+				require.Error(t, err)
+				assert.EqualError(t, err, "unable to ensure TemplateRequestUpdate resource after NSTemplateTier changed: unable to get TemplateUpdateRequest for Space 'user-0': mock error")
+				assert.Equal(t, reconcile.Result{}, res) // no explicit requeue
+			})
 		})
 
-		t.Run("unable to update NSTemplateTier status", func(t *testing.T) {
+		t.Run("unable to delete TemplateUpdateRequest associated with MasterUserRecord/Space", func(t *testing.T) {
+			t.Run("MasterUserRecords", func(t *testing.T) {
+				// TODO: this test should be removed once migration from MUR -> Spaces is completed.
 
-			t.Run("when starting update process", func(t *testing.T) {
 				// given
-				basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates)
+				previousBasicTier := tiertest.BasicTier(t, tiertest.PreviousBasicTemplates)
+				basicTier := tiertest.BasicTier(t, tiertest.CurrentBasicTemplates, tiertest.WithCurrentUpdateInProgress())
 				initObjs := []runtime.Object{basicTier}
-				initObjs = append(initObjs, murtest.NewMasterUserRecords(t, 10, "new-user-%d",
-					murtest.Account("cluster1", *basicTier))...)
+				initObjs = append(initObjs, murtest.NewMasterUserRecords(t, 10, "user-%d",
+					murtest.Account("cluster1", *previousBasicTier))...)
+				initObjs = append(initObjs, turtest.NewTemplateUpdateRequests(maxPoolSize, "user-%d", *basicTier, turtest.Complete("user-0"))...)
+
 				r, req, cl := prepareReconcile(t, basicTier.Name, initObjs...)
-				cl.MockStatusUpdate = func(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error {
-					if _, ok := obj.(*toolchainv1alpha1.NSTemplateTier); ok {
-						return fmt.Errorf("mock error")
+				cl.MockDelete = func(ctx context.Context, obj client.Object, opts ...client.DeleteOption) error {
+					if _, ok := obj.(*toolchainv1alpha1.TemplateUpdateRequest); ok {
+						return fmt.Errorf("mock error") // must not be a `NotFoundError` in this test
```

Comment:
I don't understand the comment vs the error that is returned (it's not a `NotFoundError`)

Reply:
It just means that the mock error should not make `errors.IsNotFound()` return true. But I think this comment doesn't really add much so I've just removed it in b4ebc5d377fd38f12f28fef5d9ed3966574281c6

Reply:
my apologies, I think I just misunderstood the comment when reading it yesterday :/